godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,818 @@
+        script.obj_sys()
+    }
+
+    /// # Safety
+    ///
+    /// - `ptr` is expected to point to a list with both a length and capacity of `list_length`.
+    /// - The list pointer `ptr` must have been created with [`transfer_ptr_list_to_godot`].
+    unsafe fn transfer_ptr_list_from_godot<T>(ptr: *const T, list_length: usize) -> Vec<T> {
+        Vec::from_raw_parts(ptr as *mut T, list_length, list_length)
+    }
+
+    /// # Safety
+    ///
+    /// - The engine has to provide a valid string return pointer.
+    unsafe fn transfer_string_to_godot(string: GString, return_ptr: sys::GDExtensionStringPtr) {
+        string.move_return_ptr(
+            return_ptr as sys::GDExtensionTypePtr,
+            sys::PtrcallType::Standard,
+        );
+    }
+
+    /// # Safety
+    ///
+    /// - The engine has to manage the reference once it receives the pointer, including the ref-count.
+    unsafe fn transfer_object_to_godot(object: Gd<Object>) -> sys::GDExtensionObjectPtr {
+        let raw_gd = ManuallyDrop::new(object.into_ffi());
+
+        raw_gd.sys() as sys::GDExtensionObjectPtr
+    }
+
+    fn transfer_property_state_to_godot(
+        props: Vec<(StringName, Variant)>,
+        add_prop: unsafe extern "C" fn(
+            name: sys::GDExtensionConstStringNamePtr,
+            value: sys::GDExtensionConstVariantPtr,
+            data: *mut c_void,
+        ),
+        data: *mut c_void,
+    ) {
+        props.into_iter().for_each(|(name, value)| {
+            let name = ManuallyDrop::new(name);
+            let value = ManuallyDrop::new(value);
+
+            // SAFETY: Godot expects a string name and a variant pointer for each property. After receiving the pointer, the engine is responsible
+            // for managing the memory behind those references.
+            unsafe { add_prop(name.string_sys(), value.var_sys(), data) };
+        });
+    }
+
+    fn transfer_script_lang_to_godot(
+        script_lang: Gd<ScriptLanguage>,
+    ) -> sys::GDExtensionScriptLanguagePtr {
+        script_lang.obj_sys() as sys::GDExtensionScriptLanguagePtr
+    }
+
+    /// # Safety
+    ///
+    /// - The engine has to provide valid pointers to the instance data, property name and value.
+    pub(super) unsafe extern "C" fn set_func<T: ScriptInstance>(
+        p_instance: sys::GDExtensionScriptInstanceDataPtr,
+        p_name: sys::GDExtensionConstStringNamePtr,
+        p_value: sys::GDExtensionConstVariantPtr,
+    ) -> sys::GDExtensionBool {
+        let name = transfer_string_name_from_godot(p_name);
+        let value = &*Variant::ptr_from_sys(p_value);
+        let ctx = || format!("error when calling {}::set", type_name::<T>());
+
+        let result = handle_panic(ctx, || {
+            let instance = instance_data_as_script_instance::<T>(p_instance);
+
+            borrow_instance_mut(instance).set(name.clone(), value)
+        })
+        .unwrap_or_default();
```

Comment:
I will change this to `true`. That makes more sense.

Reply:
And I confused this with the `recount_decremented_func` function. Here, the return value does indeed represent success / indicates that the script did handle property assignment. Otherwise, the engine will call the setter of the built-in type.